### PR TITLE
Create Fedora 27 to 28 upgrade guide

### PR DESCRIPTION
**Pull request by andrewdavidwong**
For QubesOS/qubes-issues#3791

### CODE REVIEW

**Comment by pgerber** [comment]
Fedora 28 failed to become ready when I tried to update using this guide. In order to get it working had I reinstalled `qubes-gui-agent` and `qubes-vm-dependencies`. I'm not sure that's the right thing to do but I noticed that they were removed during the update. So, I figured I'd give it a try.

Short recap of how I updated:

1. Updated to Fedora 27 from Fedora 26 (with testing repo)
2. From there I attempted an update to Fedora 28 using this guide
3. Update failed, so I used that `dnf swap` trick described in the guide, that worked
4. The first time, I tried to start Fedora, `qvm-run fedora-28 …` failed with a timeout error. The second time: I installed the above mentioned packages, everything worked

Some of the history in the terminal was already truncated so [this gist](https://gist.github.com/pgerber/a5dd4a34cffcd07c6cf682e1c368a2bc) starts when the second update attempt is in progress (with `-x python2-xcffib -x qubes-core-vm`)

**Comment by marmarek** [comment]
Ah, on 4.0 the command should have `-x qubes-gui-agent`, not `-x qubes-gui-vm`.

File: managing-os/templates/fedora/upgrade-27-to-28.md [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+    Error: Transaction check error:
+      file /usr/lib/python2.7/site-packages/xcffib-0.5.1-py2.7.egg-info/PKG-INFO from install of python2-xcffib-0.5.1-5.fc28.noarch conflicts with file from package python-xcffib-0.5.1-1.fc26.noarch
+      file /usr/lib/python2.7/site-packages/xcffib/_ffi.pyc from install of python2-xcffib-0.5.1-5.fc28.noarch conflicts with file from package python-xcffib-0.5.1-1.fc26.noarch
+      file /usr/lib/python2.7/site-packages/xcffib/_ffi.pyo from install of python2-xcffib-0.5.1-5.fc28.noarch conflicts with file from package python-xcffib-0.5.1-1.fc26.noarch
+      file /usr/lib/python2.7/site-packages/xcffib/xinput.pyc from install of python2-xcffib-0.5.1-5.fc28.noarch conflicts with file from package python-xcffib-0.5.1-1.fc26.noarch
+      file /usr/lib/python2.7/site-packages/xcffib/xinput.pyo from install of python2-xcffib-0.5.1-5.fc28.noarch conflicts with file from package python-xcffib-0.5.1-1.fc26.noarch
+
+To work around this error:
+
+1. Upgrade everything besides `python2-xcffib` and `qubes-core-vm` (use `-x python2-xcffib -x qubes-core-vm`).
```
Comment (marmarek) [inline managing-os/templates/fedora/upgrade-27-to-28.md]: My bad, it should be `qubes-gui-vm` not `qubes-core-vm`. And on 4.0 the package is named `qubes-gui-agent`. We can have universal version by specifying all three:
`-x python2-xcffib -x qubes-gui-vm -x qubes-gui-agent`

File: managing-os/templates/fedora/upgrade-27-to-28.md [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+        [user@fedora-28 ~]$ sudo dnf clean all
+        [user@fedora-28 ~]$ sudo dnf --releasever=28 distro-sync --best --allowerasing
+
+    **Note:** `dnf` might ask you to approve importing a new package signing
+    key. For example, you might see a prompt like this one:
+
+        warning: /var/cache/dnf/fedora-d02ca361e1b58501/packages/python2-babel-2.3.4-1.fc28.noarch.rpm: Header V3 RSA/SHA286 Signature, key ID f5282ee4: NOKEY
+        Importing GPG key 0xF5282EE4:
+         Userid     : "Fedora (28) <fedora-28-primary@fedoraproject.org>"
+         Fingerprint: 860E 19B0 AFA8 00A1 7518 81A6 F55E 7430 F528 2EE4
```
Comment (marmarek) [inline managing-os/templates/fedora/upgrade-27-to-28.md]: Fingerprint for fc28 is `128C F232 A937 1991 C8A6  5695 E08E 7E62 9DB6 2FB1`

File: managing-os/templates/fedora/upgrade-27-to-28.md [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
```
Comment (marmarek) [inline managing-os/templates/fedora/upgrade-27-to-28.md]: Fingerprint for Fedora 28 is `128C F232 A937 1991 C8A6  5695 E08E 7E62 9DB6 2FB1`

File: managing-os/templates/fedora/upgrade-27-to-28.md [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+
+ 3. Attempt the upgrade process in the new template.
+
+        [user@fedora-28 ~]$ sudo dnf clean all
+        [user@fedora-28 ~]$ sudo dnf --releasever=28 distro-sync --best --allowerasing
+
+    **Note:** `dnf` might ask you to approve importing a new package signing
+    key. For example, you might see a prompt like this one:
+
+        warning: /var/cache/dnf/fedora-d02ca361e1b58501/packages/python2-babel-2.3.4-1.fc28.noarch.rpm: Header V3 RSA/SHA286 Signature, key ID f5282ee4: NOKEY
```
Comment (pgerber) [inline managing-os/templates/fedora/upgrade-27-to-28.md]: > `RSA/SHA286`

If I believe the upgrade guides, Fedora 28 uses sha286, Fedora 27 sha276 and Fedora 26 sha266.

**Comment by andrewdavidwong** [comment]
Thanks, all. Corrections pushed.

**Pull request merged**